apps/andi/lib/andi_web/live/ingestion_live_view/metadata/metadata_form.ex

```diff
@@ -1,209 +1,112 @@
-       changeset: changeset,
-       select_dataset_modal_visibility: "hidden",
-       search_results: [],
-       search_text: "",
-       ingestion_published?: ingestion_published?,
-       selected_dataset: ingestion.targetDataset,
-       old_selected_dataset: nil,
-       ingestion_id: ingestion.id
+       select_dataset_modal_visibility: "hidden"
      )}
   end
 
   def render(assigns) do
-    ~L"""
-    <%= f = form_for @changeset, "#", [ as: :form_data, phx_change: :validate, id: :ingestion_metadata_form] %>
-      <div class="ingestion-metadata-form ingestion-metadata-form__name">
-        <%= label(f, :name, "Name", class: "label label--required") %>
-        <%= text_input(f, :name, [class: "ingestion-name input ingestion-form-fields", phx_debounce: "1000", required: true]) %>
-        <%= ErrorHelpers.error_tag(f, :name, bind_to_input: false) %>
-      </div>
-
-      <div class="ingestion-metadata-form ingestion-metadata-form__format">
-        <%= label(f, :sourceFormat, "Source Format", class: "label label--required") %>
-        <%= select(f, :sourceFormat, MetadataFormHelpers.get_source_format_options(), [class: "select ingestion-form-fields", required: true, disabled: @ingestion_published?]) %>
-        <%= ErrorHelpers.error_tag(f, :sourceFormat, bind_to_input: false) %>
-      </div>
-
-      <div class="metadata-form__top-level-selector">
-        <%= label(f, :topLevelSelector, DisplayNames.get(:topLevelSelector), class: MetadataFormHelpers.top_level_selector_label_class(input_value(f, :sourceFormat))) %>
-        <%= if input_value(f, :sourceFormat) not in ["xml", "json", "text/xml", "application/json"] do %>
-          <%= text_input(f, :emptyValue, [class: "input--text input disable-focus", readonly: true]) %>
-        <% else %>
-          <%= text_input(f, :topLevelSelector, [class: "input--text input"]) %>
-        <% end %>
-        <%= ErrorHelpers.error_tag(f, :topLevelSelector) %>
-      </div>
-
-      <div class="ingestion-metadata-form ingestion-metadata-form__target-dataset">
-        <%= label(f, :targetDatasetName, "Dataset Name", class: "label label--required") %>
-        <%= hidden_input(f, :targetDataset, value: @selected_dataset) %>
-        <%= text_input(f, :targetDatasetName, [class: "input ingestion-form-fields", value: get_dataset_name(@selected_dataset), disabled: true, required: true]) %>
-        <button class="btn btn--select-dataset-search btn--primary-outline" phx-click="select-dataset" type="button">Select Dataset</button>
-        <%= ErrorHelpers.error_tag(f, :targetDataset, bind_to_input: false) %>
-      </div>
-
-    </form>
-    <%= live_component(@socket, AndiWeb.IngestionLiveView.SelectDatasetModal, visibility: @select_dataset_modal_visibility, search_results: @search_results, search_text: @search_text, selected_dataset: @selected_dataset, id: :ingestion_metadata_search) %>
+    selected_dataset = assigns.changeset.changes
+      |> Map.get(:targetDataset, "")
 
+    ~L"""
```

Comment:
Sadly, git didn't detect diffs here. The main changes for this html:

- phx_target is added to most interactions
- (within live_components) phx_target uses @myself to send messages to itself
- If phx_target is not specified, the event will be sent to the parent process (live view)

Also, since the select_dataset_modal cannot close itself, I'm passing a delegate function so the child can determine when it should close

Reply:
Hmm. Diffs alone might be enough to convince me to switch to the "html in a template file" pattern we've seen in some of those examples elsewhere. A later effort though.